### PR TITLE
fix: Reimplemented selectors using `StatusComboBox`

### PR DESCRIPTION
**Pull request by igor-sirotin**
Requires https://github.com/status-im/StatusQ/pull/815
Required by https://github.com/status-im/status-desktop/pull/6508

### What does the PR do

Updated selector components using `StatusComboBox`:
- [x] `AddressSourceSelector` - removed, it was totally equal to default `StatusComboBox`
- [x] `ContactsSelector`

### Affected areas

wallet

### Screenshot of functionality (including design for comparison)

I've changed the behavior for an **empty** `ContactsSelector`.
Previously it was enabled and clicking on it would open a dropdown menu with "No contacts" text.
Now it's disabled with "No contacts" text as `contentItem`.

I couldn't find designs for this.
This solution seems better to me, as the user don't have to make another click to see that there're no options to select from.

- [x] I've checked the design and this PR matches it

### CODE REVIEW

File: ui/imports/shared/controls/ContactSelector.qml [hunk 1]
```diff
@@ -83,58 +84,100 @@ Item {
...
-                height: 18
                 color: Theme.palette.baseColor1
             }
         }
 
-        selectMenu.delegate: menuItem
-        selectMenu.width: dropdownWidth
+        delegate: StatusItemDelegate {
+            id: itemContainer
+            property var currentContact: Utils.getContactDetailsAsJson(pubKey)
```
Comment (caybro) [inline ui/imports/shared/controls/ContactSelector.qml:127]: readonly?

File: ui/imports/shared/controls/ContactSelector.qml [hunk 2]
```diff
@@ -83,58 +84,100 @@ Item {
...
+                        Layout.fillWidth: true
+
+                        StatusBaseText {
+                            Layout.fillWidth: true
+                            text: currentContact.name + " • "
+                            visible: currentContact.ensVerified
+                            color: Theme.palette.baseColor1
+                            font.pixelSize: 12
+                        }
+                        StatusBaseText {
```
Comment (caybro) [inline ui/imports/shared/controls/ContactSelector.qml:162]: These should really be one text, see:
https://hackmd.io/@status-desktop/B1eOaf-nd/%2FjNzZnH8ZRiynrrXJ--XpmQ#Dynamic

File: ui/imports/shared/controls/RecipientSelector.qml [hunk 3]
```diff
@@ -202,28 +209,31 @@ Item {
...
                 root.selectedRecipient = { address, name, color, assets, fiatBalance, type: RecipientSelector.Type.Account }
             }
             onIsValidChanged: root.validate()
         }
-        AddressSourceSelector {
+        StatusComboBox {
             id: selAddressSource
             visible: isSelectorVisible && !root.readOnly
-            sources: root.sources.filter(source => source.visible)
+            model: root.sources.filter(source => source.visible)
```
Comment (caybro) [inline ui/imports/shared/controls/RecipientSelector.qml:223]: This screams for SortFilterProxyModel :)

**Review by caybro — CHANGES_REQUESTED** [review]
Some minor nits, otherwise looks cool!

File: ui/imports/shared/controls/ContactSelector.qml [hunk 4]
```diff
@@ -18,10 +18,9 @@ Item {
     id: root
     property var contactsStore
     property var selectedContact
-    height: select.height
     property int dropdownWidth: width
     property string validationError: qsTr("Please select a contact")
-    property alias validationErrorAlignment: select.validationErrorAlignment
+//    property alias validationErrorAlignment: comboBox.validationErrorAlignment
```
Comment (Khushboo-dev-cpp) [inline ui/imports/shared/controls/ContactSelector.qml:23]: why is this commented?

File: ui/imports/shared/controls/RecipientSelector.qml (hunk 3 above)
Comment (caybro) [reply ui/imports/shared/controls/RecipientSelector.qml:223]: Sure

File: ui/imports/shared/controls/ContactSelector.qml (hunk 2 above)
Comment (caybro) [reply ui/imports/shared/controls/ContactSelector.qml:166]: Yeah, there's shouldn't be an explicit width limit :) and if yes, `elide` is the way to go

File: ui/i18n/qml_en.ts [hunk 5]
```diff
@@ -686,15 +686,6 @@ Assets won&#8217;t be sent yet.</source>
 Assets won&#8217;t be sent yet.</translation>
     </message>
 </context>
-<context>
```
Comment (caybro) [inline ui/i18n/qml_en.ts:689]: You don't have to touch TS files at all

**Review by caybro — APPROVED** [review]

**Review by faust4exe — APPROVED** [review]

**Review by MishkaRogachev — APPROVED** [review]
LGTM!

**Comment by caybro** [comment]
@igor-sirotin Noticed one minor issue: 

Snímek obrazovky z 2022-08-03 14-34-46

**Comment by caybro** [comment]
And fixed :)

Snímek obrazovky z 2022-08-03 14-39-14

**Comment by Khushboo-dev-cpp** [comment]
this brings back https://github.com/status-im/status-desktop/issues/6499

**Pull request merged**